llvm/test/MC/RISCV/rv32c-valid.s

```diff
@@ -147,6 +147,11 @@ c.sub a4, a5
 # CHECK-ASM: encoding: [0x01,0x00]
 # CHECK-NO-EXT:  error: instruction requires the following: 'C' (Compressed Instructions) or 'Zca' (part of the C extension, excluding compressed floating point loads/stores){{$}}
 c.nop
+# CHECK-ASM: c.addi zero, 0
```

Comment:
The spec says "C.ADDI is only valid when rd≠x0 and imm≠0".

Reply:
I know, but this syntax is supported by binutils. We have a internal tests from our hardware validation team that uses this syntax.

The isa-manual used to say "C.NOP is encoded as c.addi x0, 0 and so expands to addi x0, x0, 0. Breakpoint Instruction." but it was changed 6 years ago. https://github.com/riscv/riscv-isa-manual/pull/248

Reply:
> I know, but this syntax is supported by binutils.

Make sense to me.